dao/src/main/java/com/epam/esm/constants/ApplicationConstants.java

```diff
@@ -1,121 +1,117 @@
+
+    //GiftCertificate QueryBuilder
+    public final static String ID_FIELD = "id";
+    public final static String NAME_FIELD = "name";
+    public final static String DESCRIPTION_FIELD = "description";
+    public final static String PRICE_FIELD = "price";
+    public final static String DURATION_FIELD = "duration";
+    public final static String CREATE_DATE_FIELD = "createDate";
+    public final static String LAST_UPDATE_DATE_FIELD = "lastUpdateDate";
     public final static String NAMES_PART_KEY = "namesPart";
     public final static String DESCRIPTION_PART_KEY = "descriptionsPart";
     public final static String TAG_NAMES_KEY = "tagNames";
+    public final static String TAG_AND_NAMES_KEY = "tagNames(and)";
+    public final static String TAGS_ATTRIBUTE_NAME = "tags";
 
-    public final static String NAMES_PART_CRITERIA_BEAN_NAME = "namePartsGCCriteria";
-    public final static String DESCRIPTION_PART_CRITERIA_BEAN_NAME = "descriptionPartsGCCriteria";
-    public final static String TAG_NAMES_CRITERIA_BEAN_NAME = "tagNamesGCCriteria";
-    public final static String ALL_CRITERIA_BEAN_NAME = "allGCCriteria";
-
-    //Tag factory
-    public final static String ALL_CRITERIA_TAG_BEAN_NAME = "allTagCriteria";
-    public final static String CERTIFICATE_ID_TAG_BEAN_NAME = "certificateIdTagCriteria";
+    //Tag QueryBuilder
+    public final static String TAG_ID_FIELD = "id";
+    public final static String TAG_NAME_FIELD = "name";
     public final static String CERTIFICATE_ID_KEY = "certificateId";
+    public final static String GC_ATTRIBUTE_NAME = "giftCertificates";
 
-    //Mappers
-    public final static String DATE_PATTERN = "yyyy-MM-dd HH:mm:ss";
-    public final static String ID_RS_KEY = "id";
-
-    //GiftCertificate mapper
-    public final static String NAME_GC_RS_KEY = "name";
-    public final static String DESCRIPTION_GC_RS_KEY = "description";
-    public final static String PRICE_GC_RS_KEY = "price";
-    public final static String DURATION_GC_RS_KEY = "duration";
-    public final static String CREATE_DATE_GC_RS_KEY = "create_date";
-    public final static String LAST_UPDATE_DATE_GC_RS_KEY = "last_update_date";
+    //Order QueryBuilder
+    public final static String USER_ATTRIBUTE_NAME = "user";
 
-    //Tag mapper
-    public final static String NAME_TAG_RS_KEY = "name";
-
-    //Services
+    //QueryBuilders sorting
     public final static String SORT_FIELDS_KEY = "sortFields";
     public final static String ORDER_KEY = "order";
+    public final static String DESC_ORDER = "desc";
+    public final static String ASC_ORDER = "asc";
+
+    //QueryBuilders order
+    public final static String ORDER_ID_FIELD = "id";
+    public final static String ORDER_COST_FIELD = "cost";
+    public final static String ORDER_CREATION_DATE_FIELD = "creationDate";
 
     //Codes
     public final static int CERTIFICATE_NOT_FOUND_CODE = 40401;
     public final static int TAG_NOT_FOUND_ERROR_CODE = 40402;
```

Comment:
You have already stored codes in ResponseErrorEnum.
Why do you store them once again here?

Reply:
For default common exceptions(excluding validation exceptions) I store the codes in Enum and return the Enum to client. In order to get this Enum value i should compare codes from throwed exceptions and the codes in ResponseErrorEnum. For throwing exception with a properly code I store them as constants.

Reply:
Then I believe it would be better to replace codes in enum on these constants

Reply:
I removed two unnecessary codes and put data from ApplicationConstants in ResponseErrorEnum. Commit 47e4753.